### DISCUSSION TITLE
Vercel production build not reading environment variables for VueJS app created using VITE.

### DISCUSSION THREAD

**Discussion by manasvi2001** [vercel/vercel#6406] [Help] [answered]
As per the Vite documentation, we prefix the env vars with `VITE_` and running `vite build`, it reads the env vars using `import.meta.env.VITE_`. My deployment on vercel is not reading these variables and replacing the env vars with `undefined`. Any help would be really appreciated.

**Comment by barelyhuman** [comment, marked as answer]
Will need a little more clarity here as to what's happening, could you please answer the following. 

1. The env vars are set on your vercel portal  or a local `.env` file that the `vercel` command can read ?
2. Where are these builds run? Local? Vercel?

**Reply by manasvi2001** [reply]
So, I have the local `.env` file and in addition to that I specifically added the same variables in vercel portal as well for all the 3 environments (Production, Preview and Development). The build on vercel portal runs whenever I push the changes to main branch of my repository.

There is a catch though. If I run `vercel --prod` from my cli, the deployed build runs fine as it reads the local env vars. The same build fails to read the env vars set in portal and the deployed build doesn't work as expected.

**Reply by barelyhuman** [reply]
okay, this might need a little more check, in your build steps on the portal add an execution to a dummy shell script that you can create locally and add in the repo. 
`vercel.sh`
```sh
#!/bin/bash

echo $VARIABLE_NAME # a variable from your set variables

VARIABLE_NAME=dummy # random value to check if it's getting set 

# the actually build step to run the build.
npm run build 

```

```sh
$ chmod +x ./vercel.sh
```

and then set the build step to be `./vercel.sh`

looks unnecessary but we need to pin point if it's the portal project that's blocked and needs the variables to be re-added or if just the deployment process.

if the above returns a value before you set the value then the vite builder is what's not working properly and you can open an issue there. 
if the above returns null then you should create a another project, set the vars there only for the prod branch and check with the same method. 

Hope that helps

**Reply by manasvi2001** [reply]
Doing above returned the value before setting it.

**Reply by barelyhuman** [reply]
Cool, so it's the vite builder not reading system env variables, that can be reported as a bug or you can start a discussion there, but just so you can continue working, add the following to the shell script to test if a script created .env file can continue working with the build. 

```sh

touch .env 
echo "VARIABLE_NAME=$VARIABLE_NAME" >> .env
echo "VARIABLE_NAME_TWO=$VARIABLE_NAME_TWO" >> .env

npm run build 
```

This will basically create the .env file for you on the build step and if it is getting read and works then it's the issue from the dotenv loaded from vite, might wanna ask there. Also just to confirm, vite states that it needs the system variables to have `VITE_` prefix, so just check if you forgot to do that to the variables. 

Let me know if this works.

**Reply by manasvi2001** [reply]
This solution worked. It created a new `.env` file before the build step and the deployment works as intended. Also, my env variables had `VITE_` prefix. I'll raise a discussion in vite builder to see if there is some issue there. 

Thanks a lot. It really helped.
